### PR TITLE
Fix dispatch submission flow: simulator listing, plot resize, lint cleanup

### LINKED ISSUE #1589: Filter the Available Simulators to only show the available simulators

### LINKED ISSUE #1601: add type checking here
https://github.com/biosimulations/Biosimulations/blob/68376b254a75b3e596380f28dcd2e92a0d3ffa0d/biosimulations/apps/dispatch-api/src/simulation-run/simulation-run.controller.ts#L126-L131

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 68376b254a75b3e596380f28dcd2e92a0d3ffa0d when #1520 was merged. cc @biosimulations.

### LINKED ISSUE #1604: Use this implementation to send job
@gmarupilla Would something like be suffcient for replacing the other method and addresssing #1526?

---

https://github.com/biosimulations/Biosimulations/blob/68376b254a75b3e596380f28dcd2e92a0d3ffa0d/biosimulations/apps/dispatch-service/src/app/services/hpc/hpc.service.ts#L35-L40

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 68376b254a75b3e596380f28dcd2e92a0d3ffa0d when #1520 was merged. cc @biosimulations.

### LINKED ISSUE #1605: save the sbatch into simDirbase
https://github.com/biosimulations/Biosimulations/blob/68376b254a75b3e596380f28dcd2e92a0d3ffa0d/biosimulations/apps/dispatch-service/src/app/services/hpc/hpc.service.ts#L48-L53

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 68376b254a75b3e596380f28dcd2e92a0d3ffa0d when #1520 was merged. cc @biosimulations.

### PR DESCRIPTION
**Pull request by gmarupilla**
Closes #1589
Closes #1604 
Closes #1605 
Closes #1601

### CODE REVIEW

File: biosimulations/apps/dispatch-service/src/app/app.controller.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class AppController {
    * @param data The payload sent for the created simulation run message
    */
   @MessagePattern(DispatchMessage.created)
-  async uploadFile(data: DispatchCreatedPayload): Promise<createdResponse> {
+  async uploadFile(data: any): Promise<createdResponse> {
```
Comment (bilalshaikh42) [inline biosimulations/apps/dispatch-service/src/app/app.controller.ts]: Why is the typing removed here?

File: biosimulations/apps/dispatch-api/src/app/app.service.ts [hunk 2]
```diff
@@ -85,6 +85,12 @@ export class AppService {
     };
   }
 
+  private async getJobCancel(uuid: string) {
```
Comment (bilalshaikh42) [inline biosimulations/apps/dispatch-api/src/app/app.service.ts:29]: Is this designed to cancel a running job?

File: biosimulations/apps/dispatch-service/src/app/app.controller.ts (hunk 1 above)
Comment (bilalshaikh42) [reply biosimulations/apps/dispatch-service/src/app/app.controller.ts]: The same time should be added to the api end to send the payload

File: biosimulations/libs/shared/ui/src/lib/tab-page/tab-page-tab.component.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
     {{ heading }}
   </ng-template>
 
-  <div [ngClass]="{'partial-width': partialWidth}">
+  <div [ngClass]="{'partial-width': partialWidth}" style="height: 100%">
```
Comment (bilalshaikh42) [inline biosimulations/libs/shared/ui/src/lib/tab-page/tab-page-tab.component.html]: This change may be not possible. This is a common library across multiple apps.  We would need to check the other apps as well before making the change.  This should perhaps be done by the dispatch app using the :host selector in its own css. 

@jonrkarr  could you approve/disapprove?

**Review by bilalshaikh42 — CHANGES_REQUESTED** [review]

File: biosimulations/apps/dispatch-api/src/app/app.service.ts (hunk 2 above)
Comment (gmarupilla) [reply biosimulations/apps/dispatch-api/src/app/app.service.ts:29]: Yes, but it's still incomplete. Not properly tested

File: biosimulations/apps/dispatch-service/src/app/app.controller.ts (hunk 1 above)
Comment (gmarupilla) [reply biosimulations/apps/dispatch-service/src/app/app.controller.ts]: I wasn't able to get filename when typing was applied. When typing was removed I was getting the filename.
Maybe something was missing in type definition, couldn't exactly figure out why.

Comment (bilalshaikh42) [reply biosimulations/apps/dispatch-service/src/app/app.controller.ts]: Resolved. Was due to a typo in the filename previously

**Review by bilalshaikh42 — DISMISSED** [review]

**Review by bilalshaikh42 — DISMISSED** [review]

**Review by bilalshaikh42 — APPROVED** [review]

**Review by bilalshaikh42 — DISMISSED** [review]

**Review by bilalshaikh42 — DISMISSED** [review]

**Review by bilalshaikh42 — APPROVED** [review]

**Pull request merged**